src/token/token.service.ts

```diff
@@ -0,0 +1,41 @@
+import { HttpException, Injectable } from '@nestjs/common';
+import * as jose from 'jose';
+
+@Injectable()
+export class TokenService {
+  async decodeToken(
+    issuer: string,
+    keyMaterialEndpoint: string,
+    tokenString: string,
+  ): Promise<[string, string]> {
+    if (issuer === undefined || issuer === '') {
```

Comment:
I think it is a lot more readable if we use separate if-statements. Since I don't expect significant performance impacts, so I would use the more readable version.

Reply:
I am ok with that, if you all have got this opinion. I learned that in AlgoDat (Big-O-Notation) and HPC.